fast64_internal/oot/oot_cutscene.py

```diff
@@ -0,0 +1,604 @@
+	
+	unkType : bpy.props.StringProperty(name = '', default = '0x0001')
+	fxType : bpy.props.EnumProperty(items = ootEnumCSTransitionType)
+	fxStartFrame : bpy.props.IntProperty(name = '', default = 0, min = 0)
+	fxEndFrame : bpy.props.IntProperty(name = '', default = 1, min = 0)
+
+def drawCSListProperty(layout, listProp, listIndex, objName, collectionType):
+	layout.prop(listProp, 'expandTab', 
+		text = listProp.listType + ' List' if listProp.listType != 'FX' else 'Scene Trans FX',
+		icon = 'TRIA_DOWN' if listProp.expandTab else 'TRIA_RIGHT')
+	if not listProp.expandTab: return
+	box = layout.box().column()
+	drawCollectionOps(box, listIndex, collectionType, None, objName, False)
+	
+	if listProp.listType == "Textbox":
+		attrName = "textbox"
+	elif listProp.listType == "FX":
+		prop_split(box, listProp, 'fxType', 'Transition')
+		prop_split(box, listProp, 'fxStartFrame', 'Start Frame')
+		prop_split(box, listProp, 'fxEndFrame', 'End Frame')
+		return
+	elif listProp.listType == "Lighting":
+		attrName = "lighting"
+	elif listProp.listType == "Time":
+		attrName = "time"
+	elif listProp.listType in ["PlayBGM", "StopBGM", "FadeBGM"]:
+		attrName = "bgm"
+	elif listProp.listType == "Misc":
+		attrName = "misc"
+	elif listProp.listType == "0x09":
+		attrName = "nine"
+	elif listProp.listType == "Unk":
+		prop_split(box, listProp, 'unkType', 'Unk List Type')
+		attrName = "unk"
+	else:
+		raise PluginError("Invalid listType")
+		
+	dat = getattr(listProp, attrName)
+	for i, p in enumerate(dat):
+		p.draw(box, listProp, listIndex, i, objName, collectionType)
+	if len(dat) == 0:
+		box.label(text = "No items in " + listProp.listType + " List.")
+	if listProp.listType == "Textbox":
+		row = box.row(align=True)
+		for l in range(3):
+			addOp = row.operator(OOTCSTextboxAdd.bl_idname, text = 'Add ' + ootEnumCSTextboxType[l][1], icon = ootEnumCSTextboxTypeIcons[l])
+			addOp.collectionType = collectionType + '.textbox'
+			addOp.textboxType = ootEnumCSTextboxType[l][0]
+			addOp.listIndex = listIndex
+			addOp.objName = objName
+	else:
+		addOp = box.operator(OOTCollectionAdd.bl_idname, text = 'Add item to ' + listProp.listType + ' List')
+		addOp.option = len(dat)
+		addOp.collectionType = collectionType + '.' + attrName
+		addOp.subIndex = listIndex
+		addOp.objName = objName
+
+
+class OOTCSListAdd(bpy.types.Operator):
+	bl_idname = 'object.oot_cslist_add'
+	bl_label = 'Add CS List'
+	bl_options = {'REGISTER', 'UNDO'} 
+
+	collectionType : bpy.props.StringProperty()
+	listType : bpy.props.EnumProperty(items = ootEnumCSListType)
+	objName : bpy.props.StringProperty()
+
+	def execute(self, context):
+		collection = getCollection(self.objName, self.collectionType, None)
+		collection.add()
+		collection[len(collection)-1].listType = self.listType
+		#self.report({'INFO'}, 'Success!')
+		return {'FINISHED'} 
+
+def drawCSAddButtons(layout, objName, collectionType):
+	def addButton(row):
+		nonlocal l
```

Comment:
```suggestion
	def addButton(row, l):
```

Reply:
The change as suggested is incorrect, `l` is modified by the function. I could take `l` as a parameter and also return a modified value for it, but this is also awkward. I wouldn't use `nonlocal` for an "API-level" function, but since this is just for a function within another function, I think this is the most elegant way to do it.